pkg/recipes/terraform/execute.go

```diff
@@ -291,7 +296,7 @@ func downloadAndInspect(ctx context.Context, tf *tfexec.Terraform, options Optio
 }
 
 // getTerraformConfig initializes the Terraform json config with provided module source and saves it
-func getTerraformConfig(ctx context.Context, workingDir string, options Options) (*config.TerraformConfig, error) {
+func getTerraformConfig(ctx context.Context, workingDir string, options Options, armOptions *arm.ClientOptions) (*config.TerraformConfig, error) {
```

Comment:
@ytimocin , @vishwahiremat : checking: we're passing armOptions/*arm.ClientOptions in a lot of places in this PR.  is this needed?

Reply:
its not needed in execute, unless its need for provider work. I added this in initial PR for private repo support but removed it later.